pyogrio/tests/test_geopandas_io.py

```diff
@@ -479,29 +517,126 @@ def test_write_dataframe_gdalparams(tmp_path, naturalearth_lowres):
     assert test_withindex_index_filename.exists() is True
 
 
-def test_write_dataframe_geometry_type(tmp_path, naturalearth_lowres):
+def test_write_dataframe_promote_to_multi_geojson(tmp_path, naturalearth_lowres):
+    """Test .geojson, wich supports mixed multi and single geometries."""
+    input_gdf = read_dataframe(naturalearth_lowres)
+
+    # promote_to_multi=None (=default): no promotion for .geojson
+    output_path = tmp_path / "test_promote_None.geojson"
+    write_dataframe(input_gdf, output_path)
+
+    assert output_path.exists()
+    output_gdf = read_dataframe(output_path)
+    geometry_types = output_gdf.geometry.type.unique()
+    assert len(geometry_types) == 2
+
+    # promote_to_multi=True: force promotion
+    output_path = tmp_path / "test_promote.geojson"
+    write_dataframe(input_gdf, output_path, promote_to_multi=True)
+
+    assert output_path.exists()
+    output_gdf = read_dataframe(output_path)
+    geometry_types = output_gdf.geometry.type.unique()
+    assert len(geometry_types) == 1
+
+
+def test_write_dataframe_promote_to_multi_fgb(tmp_path, naturalearth_lowres):
+    """Test .fgb, which needs promotion to save mixed multi and single geometries."""
+    input_gdf = read_dataframe(naturalearth_lowres)
+
+    # promote_to_multi=None (=default), promotion for .fgb
+    output_path = tmp_path / "test_promote_None.fgb"
+    write_dataframe(input_gdf, output_path)
+
+    assert output_path.exists()
+    output_gdf = read_dataframe(output_path)
+    geometry_types = output_gdf.geometry.type.unique()
+    assert len(geometry_types) == 1
+    output_info = read_info(output_path)
+    assert output_info["geometry_type"] == "MultiPolygon"
+
+    # promote_to_multi=True: force promotion
+    output_path = tmp_path / "test_promote_True.fgb"
+    input_single_gdf = input_gdf[input_gdf.geom_type == "Polygon"]
+    write_dataframe(input_single_gdf, output_path, promote_to_multi=True)
+
+    assert output_path.exists()
+    output_gdf = read_dataframe(output_path)
+    geometry_types = output_gdf.geometry.type.unique()
+    assert len(geometry_types) == 1
+    output_info = read_info(output_path)
+    assert output_info["geometry_type"] == "MultiPolygon"
+
+    # promote_to_multi=False: prohibit promotion
+    output_path = tmp_path / "test_promote_False.fgb"
+    write_dataframe(input_gdf, output_path, promote_to_multi=False)
+
+    assert output_path.exists()
+    output_gdf = read_dataframe(output_path)
+    geometry_types = output_gdf.geometry.type.unique()
+    assert len(geometry_types) == 2
+    output_info = read_info(output_path)
+    assert output_info["geometry_type"] == "Unknown"
+
+
+def test_write_dataframe_geometry_type_unknown(tmp_path, naturalearth_lowres):
```

Comment:
You can use a parametrized test for this one as well, for values of `layer_geometry_type` and expected `geometry_type`

Reply:
I integrated in the previous test...